### ISSUE TITLE
OsmAndroid Overlay : «WIN DEATH» Crash when using canvas.drawCircle()

### ISSUE THREAD

**Issue by fabier** [osmdroid/osmdroid#82]
I've got a very weird bug on my application that displays a map with a blue circle on top of the map.

I have :

…
public class MapOverlay extends Overlay {
    …
    private Paint innerPaint;
    …
    public MapOverlay(Context context) {
        …
        this.innerPaint = new Paint();
        this.innerPaint.setColor(Color.argb(0x80, 0x43, 0x24, 0xa0));
        this.innerPaint.setStrokeWidth(2.0f);
        this.innerPaint.setStyle(Style.FILL);
        …
    }
    …
    @Override
    public void draw(Canvas canvas, MapView mapView, boolean shadow) {
    if (!shadow && mapView.getZoomLevel() == 18) {
        canvas.drawCircle(-19279750, -11004483, 8.0f, innerPaint);
    }
    …
}
This code fails (if you go to this location at level 18 on your map).

But if you change the last line to :

canvas.drawCircle(-19279750, -11004482, 8.0f, innerPaint);
canvas.drawCircle(-19279750, -11004484, 8.0f, innerPaint);
canvas.drawCircle(-19279749, -11004483, 8.0f, innerPaint);
canvas.drawCircle(-19279751, -11004483, 8.0f, innerPaint);
or a lot of other combinaisons, this code will work properly.

What is happening ?

In logs, I can see the following line:

15:07:39.120: I/WindowState(2330): WIN DEATH: Window{4274b490 u0 com.xyz.xyz/com.xyz.xyz.xyz}
Last info, I can also reproduce this bug when using the pair : x=-19279797, y=-11004415

Is there a rule to those pairs of values ???

https://stackoverflow.com/questions/27173115/why-is-my-android-crashing-when-i-do-canvas-drawcanvas-19279750-11004483

Thanks !

**Comment by spyhunter99** [comment]
is this still an issue for you? i'm guessing it's related to memory allocation

**Comment by fabier** [comment]
Hi, thank you for your feedback.
It's been nearly 2 years... it's still a (minor) issue, but it's the exact location for one of our clients.
I think that I might not be the only one with this issue so since it's causing a app CRASH, without any possibility to catch this error, I think you should investigate, being or not being an issue for me shouldn't be taken into consideration... what do you think ?

**Comment by spyhunter99** [comment]
Do you by chance have a more complete example? I took what you've posted and will push it to a branch. But i can't reproduce the bug, nor does anything draw for that matter.

**Comment by fabier** [comment]
I'm sorry, I don't have the code anymore... but I can try to get it back. I have to ask for it.
I'll have a look at your code too, maybe I'll be able to reproduce the bug.
I'll keep you posted.
Thank you for your time.

**Comment by fabier** [comment]
I can see that the code you posted doesn't reproduce the bug. There has been a change in map projection : bugged code (with windeath error) uses XY projection, instead of lat/long.
(Values for drawCircle are expressed in X/Y as you can see : ```canvas.drawCircle(-19279750, -11004483, 8.0f, innerPaint);```)

So : 
```mMapView.getController().setCenter(new GeoPoint(-19279750, -11004483 ));```
doesn't center map on the right point (as you may have experienced).
I had to change it to :
```mMapView.getController().setCenter(new GeoPoint(50.71838, -103.42443));``` to center it to the right point.

This also explains why you can't see the circle drawing over the map, since coordinates are not in the correct projection.

**Do you know how to switch to XY Projection?**
If you do, then you (and I) will be able to reproduce I think. (I'm sorry I don't remember the code I wrote 2 years ago).I'll try to get the source code in the meantime, because I can see that this information (projection type) is lacking in the bug description.

Keep me posted !

**Comment by fabier** [comment]
Maybe this could help : OSMDroid version used is 4.2-SNAPSHOT
It seems there is no reference to Projection or TileProvider in the source code that has a bug, so maybe something changed between 4.2 and current version (5.2) ?
Maybe the default projection has changed?
Can you have a look ?

**Comment by spyhunter99** [comment]
I'm not aware of osmdroid ever having anything other than web mercator projection. I used the new coordinates and it looks like somewhere in canada, a vegetable garden. Still not seeing any kind circle.  Any other changes needed?

**Comment by fabier** [comment]
Thank you for your feedback.
I won't have much time to work on it until next week, so I'll get back to you as soon as possible.
I think there's not much more to change, but I can't remember right now what's to change... I asked my former collegues for help on this point (trying to reproduce the bug), I'll keep you posted.
See you next week.

**Comment by spyhunter99** [comment]
So I've made some more changes, specifically the center point of the circle. When using the hard coded value, it must be drawing somewhere off screen. When using the projection to convert the values from lat/lon, it renders just fine. Also 8.0f is tiny. I bumped it up to 100f and at least it's noticeable now.

As far as the crash. I'm not entirely sure what the root cause is, but it appears to be fixed now. Reopen if you still encounter the problem after using the modified code
